### DISCUSSION TITLE
Google Map implementieren

### DISCUSSION THREAD

**Discussion by quargh** [neuefische/questions#77] [Web] [answered]
Hallo,

ich versuche eine Google Map in mein Capstone Template einzubinden. Das "Skeleton" der Map (Zoom Buttons usw) werden angezeigt, aber die Karte selbst wird nicht geladen.  Folgendes Beispiel läuft in der Sandbox und die Karte wird geladen: https://codesandbox.io/s/react-google-maps-api-editing-a-polygon-forked-gsjrgw?file=/src/index.js:136-157

In meinem Template sieht es so aus: ![error](https://user-images.githubusercontent.com/30600277/185089258-53373281-c566-47f0-9073-163b9bff4a7a.png)

Code:
```
import React, { useState, useRef, useCallback } from "react";
import ReactDOM from "react-dom";
import { LoadScript, GoogleMap } from "@react-google-maps/api";

export default function SandboxPage() {
	return <Map />;
}

function Map() {
	return (
		<div className="App">
			<LoadScript
				id="script-loader"
				googleMapsApiKey=""
				language="en"
				region="us"
			>
				<GoogleMap
					mapContainerClassName="App-map"
					center={{ lat: 52.52047739093263, lng: 13.36653284549709 }}
					zoom={12}
					version="weekly"
					on
				></GoogleMap>
			</LoadScript>
		</div>
	);
}
```

-----------------

Hat jemand eine Idee?

**Comment by fabian-hanso** [comment]
In der Sandbox hast du noch den LoadScript-Wrapper, welcher den API Key beinhaltet. Hier scheint er zu fehlen? Ich selbst habe mit der Map so noch nicht gearbeitet sondern mir immer selbst gebaut / react-google-map genutzt. Da geht das etwas anders. Der Loader wäre jetzt so das, was mir auf den ersten Blick auffällt.

**Reply by quargh** [reply]
I am sorry, ich verwende das Discussion Board zum ersten Mal. Mein halber Code wird weggeschnitten, obwohl ich es in Code Tags reinkopiert habe. Bekomme es gerade nicht editiert

**Reply by fabian-hanso** [reply]
Alles gut :) Man kann es ja trotzdem lesen. Hast du den LoadScript-Wrapper sonst mit drin?

**Reply by fabian-hanso** [reply]
Sollte es noch immer Probleme geben, ping mich gerne auch zur Not über Slack an.

**Reply by quargh** [reply]
So jetzt ist mein Code vollständig... drei Backtics waren nötig

**Comment by actyvystom** [comment]
Hallo, 
was @fabian-hanso sagt, er war schneller ;)
im Vergleich zur Sandbox hast Du in Deinem Code den LoadScript-Container nicht verwendet - ich gehe davon aus, dass er zwingend erforderlich ist, da er die `googleMapsApiKey`-Prop trägt:
```javascript
<LoadScript
        id="script-loader"
        googleMapsApiKey=""
        language="en"
        region="us"
      >
        <GoogleMap
          mapContainerClassName="App-map"
          center={{ lat: 52.52047739093263, lng: 13.36653284549709 }}
          zoom={12}
          version="weekly"
          on
        ></GoogleMap>
      </LoadScript>
```

**Reply by fabian-hanso** [reply]
Yes. Habe mir das Konstrukt mal fix grob angeschaut. Er lädt die Map auf Basis des Wrappers mit der ID "script-loader". Wenn dieser natürlich fehlt, lädt er alles außer die Map. :)

**Reply by quargh** [reply]
So jetzt ist mein Code vollständig... drei Backtics waren nötig

**Comment by quargh** [comment]
Hallo @fabian-hanso @actyvystom , nach Rücksprache mit Paul Reichetanz soll ich euch hier nochmals anpingen, also das Problem bleibt ungelöst... Falls das ein Mißverständnis war, mein LoadScript war von vornherein dabei, es wurde nur beim posten durch fehlendes Triple-Backtick weggeschnitten

**Comment by alexclaes** [comment]
Hallo @quargh. Kannst du bitte deinen Code in einem Branch in deinem repository comitten und pushen? Den Link bitte anschließend hier angeben.

Wenn man den kompletten Code hat und es innerhalb der Anwendung testen kann, lässt sich die Situation viel besser analysieren.

**Reply by quargh** [reply]
@aclaes ok habe ich gemacht

**Comment by quargh** [comment, marked as answer]
Ich habe es (wie oben beschrieben) in Codesandbox zum Laufen bekommen, und ich habe es in einer neuen React Anwendung zu Laufen bekommen: https://github.com/quargh/maps 
Nur im Capstone Template funktioniert es leider nicht. 
Hier der Link zu meinem Capstone Project https://github.com/quargh/capstone-project/tree/Container

**Reply by mpagels** [reply]
In deiner `styles.js` Datei, wo die `GlobalStyles` erstellst, verschiebe Zeile 8
![grafik](https://user-images.githubusercontent.com/54703643/185380553-fceca7b6-67c2-4872-89a1-2659a6f3ad82.png)

in den `body` selector. Dann sollte die Map zu sehen sein.
![grafik](https://user-images.githubusercontent.com/54703643/185380725-9834e67a-5906-4e48-9512-eebc92bac1ae.png)

**Reply by quargh** [reply]
Danke @mpagels das hat geklappt :-)
